pkg/kubectl/resource/builder.go

```diff
@@ -357,7 +373,7 @@ func (b *Builder) RequireNamespace() *Builder {
 
 // SelectEverythingParam
 func (b *Builder) SelectAllParam(selectAll bool) *Builder {
-	if selectAll && b.selector != nil {
+	if selectAll && (b.labelSelector != nil || len(b.fieldSelector) > 0) {
```

Comment:
I was wondering why different type for `labelSelector` vs `fieldSelector`, can we make a consistency ?

Reply:
```
labelSelector        *string
fieldSelector        string
```

@shiywang `labelSelector` is renamed from `selector`, which will be set to `labels.Everything` (a pointer) if not set.

> different type for labelSelector vs fieldSelector, can we make a consistency

New-added `fieldSelector` can be a pointer. @liggitt @smarterclayton WDYT?